pkg/operator/staticresourcecontroller/static_resource_controller.go

```diff
@@ -265,6 +278,28 @@ func (c *StaticResourceController) Sync(ctx context.Context, syncContext factory
 	errors := []error{}
 	var notFoundErrorsCount int
 	for _, conditionalManifest := range c.manifests {
+		ready, err := c.precondition(ctx, conditionalManifest.files)
+		if err != nil {
+			if _, _, updateErr := v1helpers.UpdateStatus(ctx, c.operatorClient, v1helpers.UpdateConditionFn(operatorv1.OperatorCondition{
+				Status:  operatorv1.ConditionFalse,
+				Reason:  "ErrorCheckingPrecondition",
+				Message: err.Error(),
+			})); updateErr != nil {
+				return updateErr
+			}
+			return err
+		}
+		if !ready {
+			if _, _, updateErr := v1helpers.UpdateStatus(ctx, c.operatorClient, v1helpers.UpdateConditionFn(operatorv1.OperatorCondition{
```

Comment:
I don't see a place where the condition is set to status=true

Reply:
Great spot! I added: `Type:    fmt.Sprintf("%sDegraded", c.name),`. The update is at the bottom of the `sync`.